docs/core-concepts/saturation-penalty.md

```diff
@@ -0,0 +1,80 @@
+
+## Time-based penalty accrual
+
+The protocal calculates penalty accrual using a time based system. `accruePenalties` function in Saturation.sol takes a `duration` parameter that is defined as the `delta` between the current time and the last time penalty was accrued. See the call stack in `AmmalgamPair.sol`: `accrueSaturationPenaltiesAndInterest -> mintPenalties -> accruePenalties`.
+
+## Penalty threshold
+
+Penalties only starts accruing when a position saturation exceeds 85% (`START_SATURATION_PENALTY_RATIO_IN_MAG2 = 85`). When calculating the total saturation in penalty, see `calcTotalSatAfterLeafInclusive` in Saturation.sol, the threshold leaf is checked against the maxLeaf.
+
+When `threasholdLeaf > maxLeaf` it means that the leaf that corresponds to a penalty threshold of 85% saturation is higher than any leaf containing a position. Since `maxLeaf` is defined as `tree.highestSetLeaf`, this means that no position has been set to a leaf that is higher than the threshold leaf. In other words, no position has reached the 85% saturation threshold.
+
+On the other hand, if the a position has reached/surpassed the 85% saturation threshold, `maxLeaf > thresholdLeaf`, then the system loops through the leaves between `thresholdLeaf` and `maxLeaf` and add up all of the saturation (`satInLAssets`). The sum of all the saturation is then returned as the `satInLAssetsInPenalty`, which will be used to calculate the penalty rate.
+
+## Penalty rate calculation
+
+```
+Penalty formula:
+
+penaltyRate = ((1 - u₀) × f_interest(u₁) × totalDeposits) / (saturationInPenalty)
+
+ref: ` calcSaturationPenaltyRatePerSecondInWads` in Saturation.sol
+```
+
+The penalty rate calculation is a dynamic risk pricing system. The rate will change based on `borrow utilization` and `saturation utilization`. The motivation of this design is to ensure that LPs are properly compensated and protocol health is maintained. The relationship of the utilization metrics to penalty rate is as follows:
+
+- Borrow utilization: lower utilization -> higher penalty rate.
+- Saturation utilization: higher utilization -> higher penalty rate.
+
+The formula above defines this relationship by using (1 - u₀), where `u₀ = current borrow utilization` that results in a scaling factor. For example:
+
+1. u₀ = 10% -> (1 - 10%) = 90% scaling factor
+2. u₀ = 90% -> (1 - 90%) = 10% scaling factor
+
+**Economic incentives**
+
+The inverse relationship of borrow utilization with penalty rate is essentially because the protocal needs to balance between two objectives:
+
+1. Discourage over-saturated positions.
+  - `Low` borrow utilization and `high` saturation utilizaiton = high penalty rate
+2. Maintain system stability by preventing cascading liquidation.
+  - `High` borrow utilization and `high` saturation utiliztion = low penalty rate
+
+*Lower borrow utilizaiton*
+
+When borrow utilization is low (e.g. u₀ = 10%), this means:
```

Comment:
use latex

```suggestion
When borrow utilization is low (e.g. $u_0 = 10%$), this means:
```

Reply:
correct format is $u_0 = 10/%$, otherwise the % won't show, I think we were running into this issue a couple times on call.